### ISSUE TITLE
Alternative animation API

### ISSUE THREAD

**Issue by Brendonovich** [Brendonovich/simple-obs#13]
~~Time-based:~~
```ts
runAnimation({
  subjects: {
    one: new Source(),
    two: new Filter()
  }, 
  keyframes: {
    0: {
      one: {
        x: 0,
        y: 0
      },
      two: {
        hue_shift: 0
      }
    },
    1000: {
      one: {
        x: 10,
        y: 10
      },
      two: {
        hue_shift: 180
      }
    }
  }
}})
```

**Comment by Brendonovich** [comment]
Or subject-based:
```ts
runAnimation({
  subjects: {
    one: new Source(),
    two: new Filter()
  }, 
  keyframes: {
    one: {
      0: {
        x: 0,
        y: 0
      },
      1000: {
        x: 10,
        y: 10
      }
    },
    two: {
      0: {
        hue_shift: 0
      },
      1000: {
        hue_shift: 180
      }
    }
  }
})
```

**Comment by Brendonovich** [comment]
Value-based:
```ts
runAnimation({
  subjects: {
    one: new Source(),
    two: new Filter()
  }, 
  keyframes: {
    one: {
      x: {
        0: 0,
        1000: 10
      },
      y: {
        0: 0,
        1000: 10
      }
    },
    two: {
      hue_shift: {
        0: 0,
        1000: 180
      }
    }
  }
})
```

**Comment by Brendonovich** [comment]
Passing the subjects as a separate argument 1. Removes the need for `keyframes()` and also allows for checking of duplicates in a much easier fashion, before keyframe processing even begins

**Comment by Brendonovich** [comment]
Property-based is the winner

**Comment by Villanuevand** [comment]
@Brendonovich 
Could we add typing to `Sources` and `Filter`?

**Comment by Brendonovich** [comment]
> @Brendonovich
> Could we add typing to `Sources` and `Filter`?

I was just using `Source` and `Filter` as examples/placeholders, they are both abstract classes that must be extended by implementation classes. There are types generated for them, and they will be documented for anyone who wants to make custom `Source`s and `Filter`s

**Comment by Brendonovich** [comment]
Closed by #14
